bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BTable.java

```diff
@@ -347,6 +347,18 @@ public boolean isInMemoryTable() {
         return true;
     }
 
+    /**
+     * Returns the length or the number of rows of the table if the table is not null.
+     *
+     * @return number of rows of the table
+     */
+    public int length() {
+        if (tableName == null) {
+            return -1;
+        }
+        return tableProvider.getRowCount(tableName);
```

Comment:
I think this will throw an NPE for Cursor based tables (BCursorTable) which are returned via SQL Connector actions.

Reply:
In this case, IMO we should just throw an exception similar to add, remove operations if the `length` operation is performed on a cursor table.